sw/airborne/firmwares/rotorcraft/guidance/guidance_indi.c

```diff
@@ -0,0 +1,228 @@
+struct FloatVect3 euler_cmd;
+
+struct FloatEulers guidance_euler_cmd;
+
+void guidance_indi_enter(void) {
+  filt_accelzbody = 0;
+  filt_accelzbodyd = 0;
+  filt_accelzbodydd = 0;
+  roll_filt = 0;
+  roll_filtd = 0;
+  roll_filtdd = 0;
+  pitch_filt = 0;
+  pitch_filtd = 0;
+  pitch_filtdd = 0;
+  FLOAT_VECT3_ZERO(filt_accel_ned);
+  FLOAT_VECT3_ZERO(filt_accel_ned_d);
+  FLOAT_VECT3_ZERO(filt_accel_ned_dd);
+}
+
+void guidance_indi_run(bool_t in_flight) {
+
+  //maybe include rc to add some position inputs
+  /* orientation vector describing simultaneous rotation of roll/pitch */
+//   struct FloatVect3 ov;
+//   ov.x = get_rc_roll_f();
+//   ov.y = get_rc_pitch_f();
+//   ov.z = 0.0;
+
+  guidance_indi_filter_attitude();
+  guidance_indi_filter_accel();
+
+  float pos_x_err = POS_FLOAT_OF_BFP(guidance_h.sp.pos.x) - stateGetPositionNed_f()->x; //+-ov.y/ (STABILIZATION_ATTITUDE_SP_MAX_THETA)*3.0;
+  float pos_y_err = POS_FLOAT_OF_BFP(guidance_h.sp.pos.y) - stateGetPositionNed_f()->y; //+ ov.x/ (STABILIZATION_ATTITUDE_SP_MAX_PHI)*3.0;
+
+  float speed_sp_x = pos_x_err*guidance_indi_pos_gain;
+  float speed_sp_y = pos_y_err*guidance_indi_pos_gain;
+
+//     float accel_x = (-ov.y/ (STABILIZATION_ATTITUDE_SP_MAX_THETA)*8.0 - stateGetSpeedNed_f()->x)*auto_speed_gain;
+//     float accel_y = (ov.x/ (STABILIZATION_ATTITUDE_SP_MAX_PHI)*8.0 - stateGetSpeedNed_f()->y)*auto_speed_gain;
+  sp_accel.x = (speed_sp_x - stateGetSpeedNed_f()->x)*guidance_indi_speed_gain;
+  sp_accel.y = (speed_sp_y - stateGetSpeedNed_f()->y)*guidance_indi_speed_gain;
+//   sp_accel.x = (radio_control.values[RADIO_PITCH]/9600.0)*8.0;
+//   sp_accel.y = -(radio_control.values[RADIO_ROLL]/9600.0)*8.0;
+
+  //   struct FloatMat33 Ga;
+  guidance_indi_calcG(&Ga);
+  MAT33_INV(Ga_inv, Ga);
+
+  float altitude_sp = POS_FLOAT_OF_BFP(guidance_v_z_sp);
+  float vertical_velocity_sp = guidance_indi_pos_gain*(altitude_sp - stateGetPositionNed_f()->z);
+//     float vertical_velocity_rc_euler = -(stabilization_cmd[COMMAND_THRUST]-4500.0)/4500.0*2.0;
+  float vertical_velocity_err_euler = vertical_velocity_sp - stateGetSpeedNed_f()->z;
+  sp_accel.z = vertical_velocity_err_euler*guidance_indi_speed_gain;
+
+  struct FloatVect3 a_diff = { sp_accel.x - filt_accel_ned.x, sp_accel.y -filt_accel_ned.y, 0.0};
+
+  Bound(a_diff.x, -6.0, 6.0);
+  Bound(a_diff.y, -6.0, 6.0);
+  Bound(a_diff.z, -9.0, 9.0);
+
+  MAT33_VECT3_MUL(euler_cmd, Ga_inv, a_diff);
+
+  guidance_euler_cmd.phi = roll_filt + euler_cmd.x;
+  guidance_euler_cmd.theta = pitch_filt + euler_cmd.y;
+  guidance_euler_cmd.psi = 0;//stateGetNedToBodyEulers_f()->psi;
+
+  //Bound euler angles to prevent flipping and keep upright
+  Bound(guidance_euler_cmd.phi, -0.7, 0.7);
+  Bound(guidance_euler_cmd.theta, -0.7, 0.7);
+
+  stabilization_attitude_set_setpoint_rp_quat_f(in_flight);
+
+}
+
+//low pass the accelerometer measurements with a second order filter to remove noise from vibrations
+void guidance_indi_filter_accel(void)
```

Comment:
Would probably make sense to use an already existing low pass filter from [sw/airborne/filters/low_pass_filter.h] (https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/filters/low_pass_filter.h)